File: reports/Ignatyuk/Lab5/src/Task1/Abstract/src/abct/Theatre.java
package abct;

public final class Theatre extends PublicBuilding {
    public Theatre(final String c_Name, final String c_Address) {
        super(c_Name, c_Address);
    }

    public Theatre(final String c_Name, final String c_Address, final Integer c_VisitorsCount) {
        super(c_Name, c_Address, c_VisitorsCount);
    }

    public Theatre(final String c_Name, final String c_Address, final Integer c_VisitorsCount,
            final Double c_VisitPrice) {
        super(c_Name, c_Address, c_VisitorsCount, c_VisitPrice);
    }

    public Theatre(final String c_Name, final String c_Address, final Integer c_VisitorsCount,
            final Double c_VisitPrice, final String c_WorkTime) {
        super(c_Name, c_Address, c_VisitorsCount, c_VisitPrice, c_WorkTime);
    }

    public final void f_show_performance() {
        System.out.println("Very interesting performance...");
    }
}

File: Reports/Daniil_Pozdnyakov/lab5/src/task1/Theatre.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace lab5._1._5
{
    public class Theatre : PublicBuilding
    {
        public Theatre(string name, string address) : base (name, address)
        {

        }

        public Theatre(string name, string address, int visitorsCount) : base(name, address, visitorsCount)
        {

        }

        public Theatre(string name, string address, int visitorsCount, double visitPrice) : base(name, address, visitorsCount, visitPrice)
        {

        }

        public Theatre(string name, string address, int visitorsCount, double visitPrice, string c_WorkTime) : base(name, address, visitorsCount, visitPrice, c_WorkTime)
        {

        }

        public void ShowPerformance()
        {
            Console.WriteLine("Very interesting performance...");
        }
    }
}